### ISSUE TITLE
[FEAT] Optional autoinsert of jira issue derived from branchname

### ISSUE THREAD

**Issue by sschneider-ihre-pvs** [vivaxy/vscode-conventional-commits#186]
### Is your feature request related to a problem? Please describe

I always reference the jira issue like ABC-123 manually.

### Describe the solution you'd like

It would be nice to have an option that would get the jira issue from the branchname that looks mostly like `feature/ABC-123-long-description-of-issue`

### Describe alternatives you've considered

Maybe some kind of footer list like the scope list would work also so you can save a footer and insert it on every commit again.

**Comment by vivaxy** [comment]
Hi @sschneider-ihre-pvs, thank you for your suggestion.

I am not quite sure about what you are suggesting. Please supply any use case in detail to help us understand.

Here is some quick answers:
1. If you'd like to insert a jira format footer manually which links to a jira issue, it is possible to do so right now. Please try add `ref: #ABC-123` or `ref: ABC-123` in the footer section. Actually you can right anything in the footer section.
2. If you want a tool that automatically extract the issue id from the branch name and append to the commit footer, is https://github.com/bk201-/jira-prepare-commit-msg helpful?

**Comment by sschneider-ihre-pvs** [comment]
I know that there are various ways to get the issue identifier out of the branchname. I was suggesting adding this as a feature to the extension.
```
fix(typescript): :bug: Disabled warnings for explicit any
\n
* Disabled warnings in eslinrc
* Fixed types in models
* Fixed typos
\n
JIRA #{{<jira-issue}}
or
JIRA: {{<jira-issue}}
```

so similar to breaking changes it would be nice to have something like this to have conventional commits for a group of ppl that work with jira.

@yi-Xu-0100 across the board, there are a lot of tools that can detect jira issues with plugins or extensions. For example, jenkins will add a link to the issue if it sees the jira issue key in a commit message.

**Comment by yi-Xu-0100** [comment]
@sschneider-ihre-pvs I am confused about the example. 😅

So the `JIRA #{{<jira-issue}}` or `JIRA: {{<jira-issue}}` is right? Not `ref #{{<jira-issue}}` or `ref: {{<jira-issue}}` ?

**Comment by sschneider-ihre-pvs** [comment]
the prefix actually doesn't matter, I just picked JIRA to make it clearer and and all the {{}} should be replaced with the issue key.

**Comment by kruzyk** [comment]
Yep, automatic insertion to commit message a issue nr from branch would be perfect :)
